torchgeo/datasets/fair1m.py

```diff
@@ -0,0 +1,373 @@
+        "Football Field": {"id": 32, "category": "Court"},
+        "Tennis Court": {"id": 33, "category": "Court"},
+        "Roundabout": {"id": 34, "category": "Road"},
+        "Intersection": {"id": 35, "category": "Road"},
+        "Bridge": {"id": 36, "category": "Road"},
+    }
+
+    image_root: str = "images"
+    labels_root: str = "labelXmls"
+
+    def __init__(
+        self,
+        root: str = "data",
+        transforms: Optional[Callable[[Dict[str, Tensor]], Dict[str, Tensor]]] = None,
+    ) -> None:
+        """Initialize a new FAIR1M dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+        """
+        self.root = root
+        self.transforms = transforms
+
+        files = sorted(glob.glob(os.path.join(self.root, self.labels_root, "*.xml")))
+        self.files = [read_pascal_voc(f) for f in files]
+
+    def __getitem__(self, index: int) -> Dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data and label at that index
+        """
+        image = self._load_image(index)
+        boxes, labels = self._load_target(index)
+        sample = {"image": image, "bbox": boxes, "label": labels}
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.files)
+
+    def _load_image(self, index: int) -> Tensor:
+        """Load a single image.
+
+        Args:
+            index: index to return
+
+        Returns:
+            the image
+        """
+        filename = self.files[index]["filename"]
+        path = os.path.join(self.root, self.image_root, filename)
+        with Image.open(path) as img:
+            array = np.array(img.convert("RGB"))
+            tensor: Tensor = torch.from_numpy(array)  # type: ignore[attr-defined]
+            # Convert from HxWxC to CxHxW
+            tensor = tensor.permute((2, 0, 1))
+            return tensor
+
+    def _load_target(self, index: int) -> Tuple[Tensor, Tensor]:
+        """Load the target mask for a single image.
+
+        Args:
+            index: index to return
+
+        Returns:
+            the target bounding boxes and labels
+        """
+        boxes_list = self.files[index]["points"]
+        labels_list = [
+            self.classes[label]["id"] for label in self.files[index]["labels"]
+        ]
+        boxes = torch.tensor(boxes_list).to(torch.float)  # type: ignore[attr-defined]
+        labels = torch.tensor(labels_list)  # type: ignore[attr-defined]
+        return boxes, labels
+
+    def plot(self, index: int) -> None:
```

Comment:
We should come up with a consistent API for this. Questions I have: Should the input be an index or a sample dict? If a sample dict, can it handle batches or just single instances? I think we might need to return the figure for use in a Lightning trainer. Also, I'm not sure if we want to call `plt.show()` ourselves as it seems to break some of the Windows tests. We could instead let the user handle that part.

Reply:
I think that since the dataset itself is only capable of loading a single sample, it's `plot` method should only plot a single sample. Now we could change this for the datamodule since that has the ability to load batches via dataloaders.

Reply:
I'm not sure how many users use Datasets other than to put them in a DataLoader. Especially for GeoDataset, it's pretty cumbersome to index the Dataset directly, and the DataLoader won't return the index it used. For VisionDatasets things are much simpler, but we want to ensure that both match.

Reply:
When I'm looking to train a model, I use DataModule, otherwise for any quick visualization purposes I would use a Dataset. I can refactor take as input a sample dict instead of an index, no problem there. My only concern is that when a user plots using `Dataset.plot` it won't be obvious how to plot a batch dict.

Reply:
I'm not sure I agree that plot should act the same way for VisionDataset and GeoDataset though. This probably could use more discussion.

Reply:
Refactored to take in a sample dict